pkg/controller/machinepool/nutanixactuator.go

```diff
@@ -0,0 +1,223 @@
+		cd.Spec.Platform.Nutanix.FailureDomains)
+
+	return &platform
+}
+
+// getNutanixDataDisksStorageConfig retrieves the storage configuration for a given Nutanix VM data disk.
+func (a *NutanixActuator) getNutanixDataDisksStorageConfig(dataDisk machinev1.NutanixVMDisk) *installertypesnutanix.StorageConfig {
+	if dataDisk.StorageConfig == nil {
+		return nil
+	}
+
+	storageConfig := &installertypesnutanix.StorageConfig{
+		DiskMode: dataDisk.StorageConfig.DiskMode,
+	}
+
+	if dataDisk.StorageConfig.StorageContainer == nil || dataDisk.StorageConfig.StorageContainer.UUID == nil {
+		return storageConfig
+	}
+
+	storageConfig.StorageContainer.UUID = *dataDisk.StorageConfig.StorageContainer.UUID
+	return storageConfig
+}
+
+// getNutanixDataDisksDataSource extracts the data source reference for a Nutanix VM data disk.
+func (a *NutanixActuator) getNutanixDataDisksDataSource(dataDisk machinev1.NutanixVMDisk) (*installertypesnutanix.StorageResourceReference, error) {
+	if dataDisk.DataSource == nil || dataDisk.DataSource.UUID == nil {
+		return nil, fmt.Errorf("data source must specify a UUID")
+	}
+
+	// Pass through UUID blindly, as MachineSets() only cares about UUID.
+	return &installertypesnutanix.StorageResourceReference{
+		UUID: *dataDisk.DataSource.UUID,
+	}, nil
+}
+
+// getNutanixDataDisks retrieves and constructs a list of Nutanix data disks from a MachinePool specification.
+func (a *NutanixActuator) getNutanixDataDisks(pool *hivev1.MachinePool, logger log.FieldLogger) ([]installertypesnutanix.DataDisk, error) {
+	var dataDisks []installertypesnutanix.DataDisk
+	for _, dataDisk := range pool.Spec.Platform.Nutanix.DataDisks {
+		storageConfig := a.getNutanixDataDisksStorageConfig(dataDisk)
+		dataSource, err := a.getNutanixDataDisksDataSource(dataDisk)
+		if err != nil {
+			return nil, err
+		}
+
+		disk := &installertypesnutanix.DataDisk{
+			DiskSize:         dataDisk.DiskSize,
+			DeviceProperties: dataDisk.DeviceProperties,
+			StorageConfig:    storageConfig,
+			DataSourceImage:  dataSource,
+		}
+		dataDisks = append(dataDisks, *disk)
+	}
+
+	logger.WithField("numDisks", len(dataDisks)).Info("found Nutanix data disks")
+	return dataDisks, nil
+}
+
+func (a *NutanixActuator) setUnsupportedConfigurationCondition(pool *hivev1.MachinePool, logger log.FieldLogger, err error) {
+	logger.WithError(err).Error("unsupported configuration detected")
+
+	conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+		pool.Status.Conditions,
+		hivev1.UnsupportedConfigurationMachinePoolCondition,
+		corev1.ConditionFalse,
+		"UnsupportedConfiguration",
+		err.Error(),
+		controllerutils.UpdateConditionIfReasonOrMessageChange,
+	)
+
+	if changed {
+		pool.Status.Conditions = conds
+		if updateErr := a.client.Status().Update(context.Background(), pool); updateErr != nil {
+			logger.WithError(updateErr).Error("failed to update MachinePool status with UnsupportedConfiguration condition")
```

Comment:
Usually when a status update fails we'll bubble the error up to ensure we requeue and try the update again. That's particularly important in this flow where you're explicitly *not* requeueing after this status update.